### DISCUSSION TITLE
Make an option to save dir whenever "cd" is used, and start console in this last used directory

### DISCUSSION THREAD

**Discussion by Brazyliszek** [git-for-windows/git#4325] [Ideas]
I'm not an advanced git user but I have to say I love version control systems. I use it to manage few of my personal hobby projects so I use the console.

Whenever I open Git Bash it starts in a default directory, in my case that is "/c/WINDOWS/system32". I tried to google if I can change it and they said that this is in fact changeable, but in order to change it, one have to click on the Git Bash's shortcut, go to properties, and set the target field to desired directory. 

So each time I start the console the first thing I have to do is to write "cd" to my current directory.

I wonder if it wouldn't be better to have an option to remember the last used directory whenever I cast "cd", and then when I start the console it will start in this saved dir. And since it'd be an option it shouldn't affect regular users.

Such feature would be very convenient for me.

**Comment by dscho** [comment]
Git Bash should start in your home directory by default. If it does not, try setting the environment variable `HOME` globally.

> I wonder if it wouldn't be better to have an option to remember the last used directory whenever I cast "cd", and then when I start the console it will start in this saved dir.

That assumes that there is only ever a single Git Bash window. That's not the common case.

Having said that, you can achieve something similar (with "last one wins" conflict resolution) by editing your `$HOME/.profile` to `cd` to the path stored in some other file (if that exists). That file would need to be written by your prompt, which you can extend by assigning something like `PS1="$PS1\$(echo "\$PWD" >/path/to/that/file)"`.

Note: I am not giving you a turnkey solution, this most likely will require some fiddling and tinkering. I just want to point you into a direction where you can help yourself to the feature you desire to have.
